_generated/AzureSpringCloudV0_Node20/deploymentProvider/azure-arm-spring-apps.ts

```diff
@@ -509,8 +509,18 @@ export class AzureSpringApps {
      */
     private async applyDeploymentModifications(appName: string, deploymentName: string, deploymentUpdateRequestBody, createDeployment: boolean) {
         console.log(`${createDeployment ? 'Creating' : 'Updating'} ${appName}, deployment ${deploymentName}...`);
-
-        let method = createDeployment ? 'PUT' : 'PATCH';
+        let isCustomContainer = false;
+        let deploymentResource = {};
+        if (!createDeployment) {
+            deploymentResource = await this.getDeploymentInfo(appName, deploymentName);
+            isCustomContainer = deploymentResource["properties"]["source"]["type"] == SourceType.CUSTOM_CONTAINER || deploymentUpdateRequestBody["properties"]["source"]["type"] == SourceType.CUSTOM_CONTAINER;
+        }
+        let method = (createDeployment || isCustomContainer) ? 'PUT' : 'PATCH';
+        if (isCustomContainer) {
+            const deploymentSettings = {...deploymentResource["properties"]["deploymentSettings"], ...deploymentUpdateRequestBody["properties"]["deploymentSettings"]};
+            deploymentUpdateRequestBody["properties"]["deploymentSettings"] = deploymentSettings;
+            deploymentUpdateRequestBody["sku"] = deploymentResource["sku"];
+        }
```

Comment:
It looks like the changes to all the 3 files are merely the same. Can we extract the logic to a common place and just reference it in these places?

Reply:
The two files in `_generated` folder are build result of `node make.js build --task AzureSpringCloudV0` command

Reply:
as discussed, this is added during the Node 20 migration.